vitrina/datasets/views.py

```diff
@@ -478,3 +484,70 @@ def get_success_url(self):
+    template_name = 'vitrina/datasets/statistics_graph.html'
+    paginate_by = 0
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        data = []
+        dataset_ids = []
+        status_translations = {}
+        status_graph_styles = {'INVENTORED': {'borderColor': 'black',
+                                              'backgroundColor': 'rgba(255, 179, 186, 0.9)',
+                                              'order': 3},
+                               'OPENED': {'borderColor': 'black',
+                                          'backgroundColor': 'rgba(255, 223, 186, 0.9)',
+                                          'order': 2},
+                               'STRUCTURED': {'borderColor': 'black',
+                                              'backgroundColor': 'rgba(186,225,255, 0.9)',
+                                              'order': 1}}
+
+        for dataset in context.get('object_list'):
+            dataset_ids.append(dataset.pk)
+
+        comments = Comment.objects\
+            .filter(content_type=ContentType.objects.get_for_model(Dataset), object_id__in=dataset_ids)\
+            .exclude(status__isnull=True).order_by('created')
+        statuses = comments.order_by('status').values_list('status', flat=True).distinct()
+
+        oldest_comment = comments.values_list('created', flat=True)[:1]
+        if oldest_comment:
+            labels = pd.date_range(oldest_comment[0].strftime("%Y %b-%d"),
+                                   datetime.datetime.now().strftime("%Y %b-%d"),
+                                   freq='MS').tolist()
+            for i in range(len(labels)):
+                labels[i] = _date(labels[i], "y b")
+            context['labels'] = labels
+
+        for status in Comment.STATUSES:
+            status_translations[status[0]] = status[1]
+
+        for item in statuses:
+            total = 0
+            counted_comments = comments.filter(status=item)\
+                .values('created__year', 'created__month')\
+                .annotate(count=Count('pk'))\
+                .order_by('created__year', 'created__month')
+            temp = []
+            for count in counted_comments:
+                total += count.get('count')
+                date = datetime.date(year=count.get('created__year'),
+                                     month=count.get('created__month'),
+                                     day=1)
+                temp.append({'x': _date(date, "y b"), 'y': total})
+            temp.append({'x': _date(datetime.datetime.now(), "y b"), 'y': total})
+            dict = {'label': str(status_translations[item]),
+                    'data': temp,
+                    'borderWidth': 1}
+            for style_element in status_graph_styles[item]:
+                dict[style_element] = status_graph_styles[item][style_element]
+            data.append(dict)
```

Comment:
This does not look good, this should be possible to do in a single query, now there are multiple queries, constructed from large lists of ids and queries are generated in a loop.

Move this to a servicve function and add tests. And try to implement this in a single query, so that this does not kill our servers.